specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-04-01-preview/cosmos-db.json

```diff
@@ -5202,6 +5202,296 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/services/{serviceName}": {
+      "put": {
+        "tags": [
+          "DataTransfer",
+          "SqlDedicatedGateway"
+        ],
+        "x-ms-examples": {
+          "DataTransferServiceCreate": {
+            "$ref": "./examples/CosmosDBDataTransferServiceCreate.json"
+          },
+          "SqlDedicatedGatewayServiceCreate": {
+            "$ref": "./examples/CosmosDBSqlDedicatedGatewayServiceCreate.json"
+          }
+        },
+        "operationId": "Service_Create",
+        "description": "Creates a service.",
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/serviceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "servicesCreateUpdateParameter",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ServiceResource"
```

Comment:
a new object called ServicesCreateUpdateParameters needs to be defined. example https://github.com/amisi01/azure-rest-api-specs/blob/dd91f643381825b4d70f259d54d9593ceaa7b61b/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-04-01-preview/cosmos-db.json#L7029

You can either use ServiceResource or create a new object which would contain name, size, type, and instance count (i.e the properties required to create a service)

The reason to decouple ServicesResource and CreateUpdateParameters is the fact that ServicesReosurce would contain readonly properties like creationTime, endpoints(in case of sqlDedicatedGateway)

Reply:
Important is to not miss   https://github.com/amisi01/azure-rest-api-specs/blob/dd91f643381825b4d70f259d54d9593ceaa7b61b/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-04-01-preview/cosmos-db.json#L7039

Reply:
There is a support to add a readonly flag in parameters. I have added it for all which are only returned.

Reply:
But it's fine, We can create a separate one as we are using discriminator.